src/jit/lower.cpp

```diff
@@ -1175,12 +1184,29 @@ void Lowering::LowerArg(GenTreeCall* call, GenTreePtr* ppArg)
     {
 
 #ifdef _TARGET_ARMARCH_
+        // We need to handle double register arguments specially on SoftFP
+        // (secondReg is only for `double` type argument on SoftFP)
+        bool      isSoftFPDoubleRegArg = false;
+        regNumber secondReg            = (info->regNum != REG_STK) ?  REG_NEXT(info->regNum) : REG_STK;
+
         // For vararg call, reg args should be all integer.
         // Insert a copy to move float value to integer register.
         if ((call->IsVarargs() || comp->opts.compUseSoftFP) && varTypeIsFloating(type))
         {
-            var_types  intType = (type == TYP_DOUBLE) ? TYP_LONG : TYP_INT;
-            GenTreePtr intArg  = comp->gtNewOperNode(GT_COPY, intType, arg);
+            isSoftFPDoubleRegArg = (comp->opts.compUseSoftFP && type == TYP_DOUBLE && info->regNum != REG_STK);
+            var_types  intType   = (type == TYP_DOUBLE) ? TYP_LONG : TYP_INT;
+
+#ifdef _TARGET_ARM_
+            // We need gtOtherRegs[] for the case - copy from a double register to two int registers
+            GenTreePtr intArg = new (comp, GT_COPY) GenTreeCopyOrReload(GT_COPY, intType, arg);
```

Comment:
> It looks like with FEATURE_MULTIREG_RET, there is an additional field. In gentree.h

Right. What I meant was that we should always use `GenTreeCopyOrReload` here even without `FEATURE_MULTIREG_RET`, in which case there is no additional cost relative to `GenTreeUnOp`, and it ensures oper/struct consistency for the `GenTree` nodes.